### PR TITLE
Add jetty-12.0 relevant content to README.md

### PR DESCRIPTION
**Pull request by steverao**
Add jetty-12.0 relevant content to README.md

### CODE REVIEW

**Review by trask — APPROVED** [review]
thanks!

File: instrumentation/jetty/README.md [hunk 1]
```diff
@@ -5,8 +5,8 @@ Jetty support is divided into the following sub-modules:
...
   the `javaagent` modules of all supported Jetty versions
 - `jetty-8.0:javaagent` applies Jetty request handler instrumentation for versions `[8, 11)`
-- `jetty-11.0:javaagent` applies Jetty request handler instrumentation for versions `[11,)`
+- `jetty-11.0:javaagent` applies Jetty request handler instrumentation for versions `[11,12)`
+- `jetty-12.0:javaagent` applies Jetty request handler instrumentation for versions `[12,)`
 
-Instrumentations in `jetty-8.0` and `jetty-11.0` are mutually exclusive, this is guaranteed by the
-instrumentation requiring parameters with types from packages `javax.servlet` or `jakarta.servlet`
-respectively.
+Instrumentations in `jetty-8.0`, `jetty-11.0` and `jetty-12.0` are mutually exclusive, this is guaranteed by the
```
Comment (laurit) [inline instrumentation/jetty/README.md:11]: I would keep the original
> Instrumentations in `jetty-8.0` and `jetty-11.0` are mutually exclusive, this is guaranteed by the
instrumentation requiring parameters with types from packages `javax.servlet` or `jakarta.servlet`
respectively.

and add an extra sentence. `jetty-12.0` instrumentation applies to a class that does not exist in earlier versions.

Comment (trask) [reply instrumentation/jetty/README.md:11]: maybe just remove the sentence altogether? doesn't seem important for end users, and is already implied by above version ranges

File: instrumentation/jetty/README.md [hunk 2]
```diff
@@ -5,8 +5,8 @@ Jetty support is divided into the following sub-modules:
...
+instrumentation requiring different parameter list respectively.
```
Comment (laurit) [inline instrumentation/jetty/README.md:12]:
```suggestion
```

**Review by laurit — APPROVED** [review]

**Pull request merged**